### PR TITLE
Implements difference() for Index & MultiIndex

### PR DESCRIPTION
**Pull request by itholic**
This PR proposes for implementing difference for Index & MultiIndex.
(https://pandas.pydata.org/docs/reference/api/pandas.Index.difference.html#pandas.Index.difference)

- Index.difference()
```python
>>> idx1 = ks.Index([2, 1, 3, 4])
>>> idx2 = ks.Index([3, 4, 5, 6])
>>> idx1.difference(idx2)
Int64Index([1, 2], dtype='int64')
>>> idx1.difference(idx2, sort=False)
Int64Index([2, 1], dtype='int64')
```

- MultiIndex.difference()
```python
>>> midx1 = ks.MultiIndex.from_tuples([('a', 'x', 1), ('b', 'y', 2), ('c', 'z', 3)])
>>> midx2 = ks.MultiIndex.from_tuples([('a', 'x', 1), ('b', 'z', 2), ('k', 'z', 3)])
>>> midx1.difference(midx2)
MultiIndex([('b', 'y', 2),
            ('c', 'z', 3)],
           )
```

### CODE REVIEW

File: databricks/koalas/indexes.py [hunk 1]
```diff
@@ -1391,6 +1391,69 @@ def set_names(self, names, level=None, inplace=False):
...
+        if not is_list_like(other):
+            raise TypeError("Input must be Index or array-like")
+        if not isinstance(sort, (type(None), type(True))):
+            raise ValueError(
+                "The 'sort' keyword only takes the values of None or True; {} was passed.".format(
+                    sort
+                )
+            )
+        # Handling MultiIndex
+        if isinstance(self, ks.MultiIndex):
```
Comment (HyukjinKwon) [inline databricks/koalas/indexes.py]: This two `if`s can be combined.

Comment (itholic) [reply databricks/koalas/indexes.py]: fixed it. thanks for the review!

**Comment by HyukjinKwon** [comment]
I am going to manually merge this to keep the author

File: databricks/koalas/indexes.py [hunk 2]
```diff
@@ -1391,6 +1391,75 @@ def set_names(self, names, level=None, inplace=False):
...
+        MultiIndex
+
+        >>> midx1 = ks.MultiIndex.from_tuples([('a', 'x', 1), ('b', 'y', 2), ('c', 'z', 3)])
+        >>> midx2 = ks.MultiIndex.from_tuples([('a', 'x', 1), ('b', 'z', 2), ('k', 'z', 3)])
+        >>> midx1.difference(midx2)  # doctest: +SKIP
+        MultiIndex([('b', 'y', 2),
+                    ('c', 'z', 3)],
+                   )
+        """
+        if not is_list_like(other):
```
Comment (HyukjinKwon) [inline databricks/koalas/indexes.py:1430]: I think we should check if `other` is `Index` instance. `is_list_like` is pandas' and we should rely on that.

File: databricks/koalas/indexes.py [hunk 3]
```diff
@@ -1391,6 +1391,75 @@ def set_names(self, names, level=None, inplace=False):
...
+            raise TypeError("Input must be Index or array-like")
+        if not isinstance(sort, (type(None), type(True))):
+            raise ValueError(
+                "The 'sort' keyword only takes the values of None or True; {} was passed.".format(
+                    sort
+                )
+            )
+        # Handling MultiIndex
+        if isinstance(self, ks.MultiIndex) and not isinstance(other, ks.MultiIndex):
+            if not all([isinstance(item, tuple) for item in other]):
```
Comment (HyukjinKwon) [inline databricks/koalas/indexes.py:1440]: Here it does a for loop which collects everything into driver side. We should probably avoid.

Comment (HyukjinKwon) [reply databricks/koalas/indexes.py:1440]: Actually, is this condition tested? doing a for loop against Koalas instance will result in something like:

```
pyspark.pandas.exceptions.PandasNotImplementedError: The method `pd.Index.__iter__()` is not implemented. If you want to collect your data as an NumPy array, use 'to_numpy()' instead.
```

**Pull request closed without merging**